src/registry/mod.rs

```diff
@@ -14,68 +14,151 @@
 
 use std::{
     fmt::{self, Display},
-    ops::{Deref, DerefMut},
     str::FromStr,
 };
 
 mod artifactory;
 #[cfg(test)]
 mod cache;
 
-pub use artifactory::Artifactory;
-use miette::{ensure, miette, Context, IntoDiagnostic};
+use crate::manifest::DependencyManifest;
+use crate::{config, manifest::Dependency};
+pub use artifactory::{Artifactory, CertValidationPolicy};
+use miette::{ensure, miette, IntoDiagnostic};
 use semver::VersionReq;
 use serde::{Deserialize, Serialize};
 use thiserror::Error;
 use url::Url;
 
-use crate::manifest::{Dependency, DependencyManifest};
-
 /// A representation of a registry URI
-#[derive(Debug, Clone, Hash, Serialize, Deserialize, PartialEq, Eq, PartialOrd, Ord)]
-pub struct RegistryUri(Url);
+#[derive(Debug, Clone, Hash, PartialEq, Eq, PartialOrd, Ord)]
+pub enum RegistryUri {
+    /// A URL to a registry
+    Url(Url),
+    /// An alias to a registry
+    Alias(String),
+    /// A resolved alias to a registry
+    ResolvedAlias {
+        /// The alias
+        alias: String,
+        /// The resolved URL
+        url: Url,
+    },
+}
+
+impl RegistryUri {
+    /// Get the raw URL of the registry with any alias resolved
+    ///
+    /// # Arguments
+    /// * `config` - The configuration to use to resolve the alias
+    pub fn with_alias_resolved(&self, config: Option<&config::Config>) -> miette::Result<Self> {
+        match self {
+            RegistryUri::Alias(alias) => match config {
+                Some(config) => {
+                    let url = config.lookup_registry(alias)?;
+                    Ok(RegistryUri::ResolvedAlias {
+                        alias: alias.clone(),
+                        url: url.clone(),
+                    })
+                }
+                None => Err(miette!("no configuration provided to resolve alias")),
```

Comment:
Please give the user the information on what alias is affected

Reply:
24b728d737a1807a6d09af46915a2829b6a9335b